frontend/lib/src/dataframes/Quiver.ts

```diff
@@ -875,6 +876,26 @@ but was expecting \`${JSON.stringify(expectedIndexTypes)}\`.
       .format(timeInSeconds % 1 === 0 ? "HH:mm:ss" : "HH:mm:ss.SSS")
   }
 
+  private static formatDecimal(value: Uint32Array, scale: number): string {
+    // This code is partly based on: https://github.com/apache/arrow/issues/35745
+    let numString = util
+      .bigNumToString(new util.BN(value))
+      .padStart(scale, "0")
+
+    if (scale === 0) {
+      return numString
```

Comment:
I'm comparing this to the original inspiration code. Why does the sign not matter here?

Reply:
ArrowJS 13 shipped with some improvements, one of them being that `bigNumToString` can actually deal with negative numbers. But it still isn't able to deal with the fractional part correctly. That's why at scale===0 we can  just return the formatted string.

Reply:
I added some comments to make that clearer